aptos-move/framework/aptos-framework/sources/account.spec.move

```diff
@@ -81,8 +81,24 @@ spec aptos_framework::account {
     }
 
     spec assert_valid_signature_and_get_auth_key(scheme: u8, public_key_bytes: vector<u8>, signature: vector<u8>, challenge: &RotationProofChallenge): vector<u8> {
-        // TODO: complex aborts conditions.
-        pragma aborts_if_is_partial;
+        aborts_if scheme == ED25519_SCHEME && len(public_key_bytes) != ed25519::PUBLIC_KEY_NUM_BYTES;
+        aborts_if scheme == ED25519_SCHEME && len(signature) != ed25519::SIGNATURE_NUM_BYTES;
+        aborts_if scheme == ED25519_SCHEME && !ed25519::spec_signature_verify_strict_t(
+            ed25519::Signature { bytes: signature },
+            ed25519::UnvalidatedPublicKey { bytes: public_key_bytes },
+            challenge
+        );
+        aborts_if scheme == MULTI_ED25519_SCHEME && (
```

Comment:
The fact that the spec of `account.move` has to be aware of all of these digital-signature related details (e.g., PK size), seems to break any reasonable notion of modularity.

Can we avoid this?

After all, these conditions were already encoded in the specs of `ed25519.move` and `multi_ed25519.move`

Reply:
@alinush , I made it modular. Please let me know how that looks to you. Thank you!